include/drivers/can.h

```diff
@@ -184,8 +188,8 @@ struct zcan_frame {
 	u8_t dlc;
 	/** The message data*/
 	union {
-		u8_t data[8];
-		u32_t data_32[2];
+		u8_t data[CAN_MAX_DLEN];
+		u32_t data_32[CAN_MAX_DLEN / sizeof(u32_t)];
```

Comment:
Calculating the size like this does not allow access to the whole data via `data_32` if `CAN_MAX_DLEN` is not a multiple of `sizeof(u32_t)` i.e. 4 bytes. Might make sense to round up here.